### PR TITLE
Allow animations with `run_time=0` and implement convenience `Add` animation

### LINKED ISSUE #4004: Please remove or weaken run_time validation in Animation class, so as to allow animations with run_time = 0
A validation on the value of run_time in the class of Animation has been recently introduced. In the current version I have this is implemented inside the begin method, but this might have already been moved in recent versions.

The problem with this validation is that it does not allow animations with run_time = 0. 

For example, with run_time = 0 one could make an object appear instantly. Note that it is not always possible to implement this example behaviour using the add method from Scene: it is conceivable that, sometimes, one may want to have a proper animation object for this, which is then combined with other animations.  

Also, having run_time = 0 does not really create any problem with the animation rendering, so, strictly speaking, the validation is not needed. My current ugly solution for this is to override the method begin to bypass this validation.

### PR DESCRIPTION
**Pull request by chopan050**
Closes #4004 

This PR refactors the `Animation.run_time` validation. Instead of checking in `Animation.begin()` that each individual animation has a positive `run_time` which is large enough for the current frame rate, we now have the following:
- `Animation.run_time` is now a property that only checks that the given value is not negative. A `run_time` of 0 is allowed.
- `validate_run_time`, originally a function defined in `manim.animation.animation`, is now a `Scene` class method, because it's now called only inside other `Scene` methods. Instead of it being called inside `Animation.begin()` to validate the `run_time` of a single animation, it's now called inside `Scene.get_run_time()` to validate the total obtained run time. This is the one which is not allowed to be 0. If it's too small for the current frame rate, it will be enlarged. Individual animations will not be affected. 

This PR also introduces a convenience animation: `Add`. It is similar to `Scene.add`, but it's an animation which can be grouped inside other animations and has a default `run_time` of 0. Thus, if people needs to instantly add `Mobject`s in the middle of an animation, they can now do so in an easier way.

Examples for using `Add` are also provided in the documentation: https://manimce--4017.org.readthedocs.build/en/4017/reference/manim.animation.animation.Add.html

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/animation/animation.py [hunk 1]
```diff
@@ -144,6 +144,7 @@ def __init__(
         **kwargs,
     ) -> None:
         self._typecheck_input(mobject)
+        self._run_time: float = run_time
         self.run_time: float = run_time
```
Comment (behackl) [inline manim/animation/animation.py:148]: Do we really need to set the actual attribute and the property here?

File: manim/animation/animation.py [hunk 2]
```diff
@@ -638,7 +624,91 @@ def __init__(
...
+
+class Add(Animation):
+    """Add Mobjects to a scene, without animating them in any other way. This
+    is similar to the :meth:`.Scene.add` method, but :class:`Add` is an
+    animation which can be grouped into other animations.
+
+    Parameters
+    ----------
+    mobjects
+        One :class:`Mobject` or more to add to a scene.
```
Comment (behackl) [inline manim/animation/animation.py:650]:
```suggestion
        One :class:`.Mobject` or more to add to a scene.
```

**Review by behackl — APPROVED** [review]
Looks good to me, thank you very much!

File: manim/animation/animation.py (hunk 1 above)
Comment (chopan050) [reply manim/animation/animation.py]: Not really. I just removed that line.

**Pull request merged**